static/js/settings_user_groups.js

```diff
@@ -14,29 +14,67 @@ exports.reload = function () {
     if (!meta.loaded) {
         return;
     }
-
     var user_groups_section = $('#user-groups').expectOne();
     user_groups_section.html('');
     exports.populate_user_groups();
 };
 
 exports.can_edit = function (group_id) {
-   var me = people.get_person_from_user_id(people.my_current_user_id());
-   return (user_groups.is_member_of(group_id, people.my_current_user_id()) || me.is_admin);
+   return (user_groups.is_member_of(group_id, people.my_current_user_id()));
 };
 
 exports.populate_user_groups = function () {
 
     var user_groups_section = $('#user-groups').expectOne();
     var user_groups_array = user_groups.get_realm_user_groups();
-    _.each(user_groups_array, function (data) {
-        user_groups_section.append(templates.render('admin_user_group_list', {
-            user_group: {
-                name: data.name,
-                id: data.id,
-                description: data.description,
-            },
-        }));
+    var me = people.get_person_from_user_id(people.my_current_user_id());
+    // We first separate out groups based on membership.
+    var editable_groups = []; // Groups of which user is a member
+    var non_editable_groups = []; // Groups of which user is not a member
+
+    _.each(user_groups_array, function (user_group) {
+        if (exports.can_edit(user_group.id)) {
+            editable_groups.push(user_group);
+        } else {
+            non_editable_groups.push(user_group);
+        }
+    });
+    // Save the length of user groups which user is member
+    var editable_group_length = editable_groups.length;
+
+    // We do the following for admin, since he/she can edit all groups.
+    if (me.is_admin) {
+        editable_groups = editable_groups.concat(non_editable_groups);
+        non_editable_groups = [];
+    }
+
+    // At this moment, editable groups mean which can be edited and
+    // non-editable groups mean which cannot be edited.
+
+    if (editable_group_length > 0) {
+        user_groups_section.append(i18n.t('<h3>Your groups</h3>'));
+    } else if (user_groups_array.length > 0) {
+        user_groups_section.append(i18n.t('<h3>Your groups</h3>'));
+        user_groups_section.append(i18n.t('<p>You are not a part of any groups.</p>'));
+    }
+
+    function append_user(user, pills) {
+        user_pill.append_person({
+            pill_widget: pills,
+            person: user,
+        });
+    }
+
+    _.each(editable_groups, function (data) {
+        // 'is_non_editable_first' parameter will hold only (if at all) in case of admins, when
+        // we have groups (for 1st such group) of which admin is not a member of.
+        var args = {
+            name: data.name,
+            id: data.id,
+            description: data.description,
+            is_non_member_first: editable_groups.indexOf(data) === editable_group_length,
```

Comment:
I could have done this in JS as well without passing that to template, but it was making the tests quite complicated, this condition will hold in case of admins only, when there exists a group of which admin is not a member. Further, I removed 'user_group' context that was being sent as arguments to templates for two reasons - 
1) To avoid linting errors
2) It was not adding any special information.
Hoping that makes some sense. :)